app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,146 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    private val homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+
+    private val popularCategories = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ON_TV,
+        MovieCategory.POPULAR_FOR_RENT,
+        MovieCategory.POPULAR_IN_THEATRES
+    )
+
+    private val nowPlayingCategories = listOf(
+        MovieCategory.NOW_PLAYING_MOVIES,
+        MovieCategory.NOW_PLAYING_TV
+    )
+
+    private val upcomingCategories = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THIS_WEEK
+    )
+
+    private val popularSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.POPULAR_STREAMING)
+    private val upcomingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.UPCOMING_TODAY)
+    private val nowPlayingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.NOW_PLAYING_TV)
+
+    private val _popularMoviesHomeViewState = MutableStateFlow(
+        HomeMovieCategoryViewState(
+            emptyList(),
+            emptyList()
+        )
+    )
+    val popularMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularMoviesHomeViewState.asStateFlow()
+
+    private val _nowPlayingMoviesHomeViewState = MutableStateFlow(
+        HomeMovieCategoryViewState(
+            emptyList(), emptyList()
+        )
+    )
+    val nowPlayingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _nowPlayingMoviesHomeViewState.asStateFlow()
+
+    private val _upcomingMoviesHomeViewState = MutableStateFlow(
+        HomeMovieCategoryViewState(
+            emptyList(),
+            emptyList()
+        )
+    )
+    val upcomingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _upcomingMoviesHomeViewState.asStateFlow()
+
+    init {
+        getPopularMovies()
+        getUpcomingMovies()
+        getNowPlayingMovies()
+    }
+
+    private fun getPopularMovies() {
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
```

Comment:
The problem here is that you are not observing popularSelected (same problem with upcoming and nowPlaying), so it won't be updated when flow emits new value (in your case selected tab is updated because you are setting the `selectedCategory` and then callling `get__Movies` method which updates view State, which is not somethign we want to do) .  We would like to observe category and movies from repo in one place and then every time one of those flows emit we want to change our viewState
So you would have to do something like this:
```
movieRepository.popMovies().collect { movies -> 
   popularSelected.collect { selected -> 
     ....
     }
     }
```
But you won't do this since we don't want to collect flow just to change value. We don't have to collect flow until we wish to do something with it (this will usually be in Composable function), we just have to change it and we have plenty of extension functions which can do this for us
So you will have to do something like this:
``` selected.flatMapLatest { repo.movies().map{ ...}}.stateIn(...)```